@wentinghu Looks like there are some test failures of the form:

```
        resp_status = requests.get(url="http://localhost:%d/version" % self._port)
        version = resp_status.text
        _logger.info(f"mlflow server version {version}")
        if version != mlflow.__version__:
>           raise Exception("version path is not returning correct mlflow version")
E           Exception: version path is not returning correct mlflow version
```

Let me know if you need help debugging.

--- a/mlflow/java/scoring/src/main/java/org/mlflow/sagemaker/ScoringServer.java
+++ b/mlflow/java/scoring/src/main/java/org/mlflow/sagemaker/ScoringServer.java
@@ -55,7 +55,8 @@ public class ScoringServer {
 
     ServletContextHandler rootContextHandler = new ServletContextHandler(null, "/");
     rootContextHandler.addServlet(new ServletHolder(new ScoringServer.PingServlet()), "/ping");
-    rootContextHandler.addServlet(new ServletHolder(new ScoringServer.VersionServlet()), "/version");
+    rootContextHandler.addServlet(new ServletHolder(
+        new ScoringServer.VersionServlet()), "/version");
     rootContextHandler.addServlet(
         new ServletHolder(new ScoringServer.InvocationsServlet(predictor)), "/invocations");
     this.server.setHandler(rootContextHandler);
@@ -171,7 +172,7 @@ public class ScoringServer {
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
       response.setStatus(HttpServletResponse.SC_OK);
-      response.getWriter().print("1.29.1.dev0");
+      response.getWriter().print("1.29.1-SNAPSHOT");
       response.getWriter().close();
     }
   }

--- a/mlflow/java/scoring/src/main/java/org/mlflow/sagemaker/ScoringServer.java
+++ b/mlflow/java/scoring/src/main/java/org/mlflow/sagemaker/ScoringServer.java
@@ -55,6 +55,7 @@ public class ScoringServer {
 
     ServletContextHandler rootContextHandler = new ServletContextHandler(null, "/");
     rootContextHandler.addServlet(new ServletHolder(new ScoringServer.PingServlet()), "/ping");
+    rootContextHandler.addServlet(new ServletHolder(new ScoringServer.VersionServlet()), "/version");
     rootContextHandler.addServlet(
         new ServletHolder(new ScoringServer.InvocationsServlet(predictor)), "/invocations");
     this.server.setHandler(rootContextHandler);
@@ -163,6 +164,15 @@ public class ScoringServer {
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) {
       response.setStatus(HttpServletResponse.SC_OK);
+    }
+  }
+
+  static class VersionServlet extends HttpServlet {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      response.setStatus(HttpServletResponse.SC_OK);
+      response.getWriter().print("1.29.1.dev0");
+      response.getWriter().close();
     }
   }
 

--- a/mlflow/java/scoring/src/test/java/org/mlflow/ScoringServerTest.java
+++ b/mlflow/java/scoring/src/test/java/org/mlflow/ScoringServerTest.java
@@ -68,6 +68,21 @@ public class ScoringServerTest {
   }
 
   @Test
+  public void testScoringServerWithValidPredictorRespondsToVersionCorrectly() throws IOException {
+    TestPredictor validPredictor = new TestPredictor(true);
+    ScoringServer server = new ScoringServer(validPredictor);
+    server.start();
+
+    String requestUrl = String.format("http://localhost:%d/version", server.getPort().get());
+    HttpGet getRequest = new HttpGet(requestUrl);
+    HttpResponse response = httpClient.execute(getRequest);
+    Assert.assertEquals(HttpServletResponse.SC_OK, response.getStatusLine().getStatusCode());
+    String responseBody = getHttpResponseBody(response);
+    Assert.assertEquals("1.29.1.dev0", responseBody);
+    server.stop();
+  }
+
+  @Test
   public void testConstructingScoringServerFromInvalidModelPathThrowsException() {
     String badModelPath = "/not/a/valid/path";
     try {

--- a/tests/helper_functions.py
+++ b/tests/helper_functions.py
@@ -248,12 +248,14 @@ def _start_scoring_proc(cmd, env, stdout=sys.stdout, stderr=sys.stderr):
 
 
 class RestEndpoint:
-    def __init__(self, proc, port, activity_polling_timeout_seconds=250):
+    def __init__(self, proc, port, activity_polling_timeout_seconds=250, validate_version=True):
         self._proc = proc
         self._port = port
         self._activity_polling_timeout_seconds = activity_polling_timeout_seconds
+        self._validate_version = validate_version
 
     def __enter__(self):
+        ping_status = None
         for i in range(self._activity_polling_timeout_seconds):
             assert self._proc.poll() is None, "scoring process died"
             time.sleep(1)
@@ -265,15 +267,16 @@ class RestEndpoint:
                     break
             except Exception:
                 _logger.info(f"connection attempt {i} failed, server is not up yet")
-        if ping_status.status_code != 200:
+        if ping_status is None or ping_status.status_code != 200:
             raise Exception("ping failed, server is not happy")
         _logger.info(f"server up, ping status {ping_status}")
 
-        resp_status = requests.get(url="http://localhost:%d/version" % self._port)
-        version = resp_status.text
-        _logger.info(f"mlflow server version {version}")
-        if version != mlflow.__version__:
-            raise Exception("version path is not returning correct mlflow version")
+        if self._validate_version:
+            resp_status = requests.get(url="http://localhost:%d/version" % self._port)
+            version = resp_status.text
+            _logger.info(f"mlflow server version {version}")
+            if version != mlflow.__version__:
+                raise Exception("version path is not returning correct mlflow version")
         return self
 
     def __exit__(self, tp, val, traceback):

--- a/tests/helper_functions.py
+++ b/tests/helper_functions.py
@@ -210,7 +210,10 @@ def pyfunc_serve_and_score_model(
     if extra_args is not None:
         scoring_cmd += extra_args
     proc = _start_scoring_proc(cmd=scoring_cmd, env=env, stdout=stdout, stderr=stdout)
-    return _evaluate_scoring_proc(proc, port, data, content_type, activity_polling_timeout_seconds)
+    validate_version = "--enable-mlserver" not in extra_args
+    return _evaluate_scoring_proc(
+        proc, port, data, content_type, activity_polling_timeout_seconds, validate_version
+    )
 
 
 def _get_mlflow_home():
@@ -317,12 +320,16 @@ class RestEndpoint:
         return response
 
 
-def _evaluate_scoring_proc(proc, port, data, content_type, activity_polling_timeout_seconds=250):
+def _evaluate_scoring_proc(
+    proc, port, data, content_type, activity_polling_timeout_seconds=250, validate_version=True
+):
     """
     :param activity_polling_timeout_seconds: The amount of time, in seconds, to wait before
                                              declaring the scoring process to have failed.
     """
-    with RestEndpoint(proc, port, activity_polling_timeout_seconds) as endpoint:
+    with RestEndpoint(
+        proc, port, activity_polling_timeout_seconds, validate_version=validate_version
+    ) as endpoint:
         return endpoint.invoke(data, content_type)
 
 

--- a/tests/models/test_cli.py
+++ b/tests/models/test_cli.py
@@ -518,7 +518,7 @@ def test_build_docker_without_model_uri(iris_data, sk_model, tmp_path):
 
 
 def _validate_with_rest_endpoint(scoring_proc, host_port, df, x, sk_model, enable_mlserver=False):
-    with RestEndpoint(proc=scoring_proc, port=host_port) as endpoint:
+    with RestEndpoint(proc=scoring_proc, port=host_port, validate_version=False) as endpoint:
         for content_type in [CONTENT_TYPE_JSON_SPLIT_ORIENTED, CONTENT_TYPE_CSV, CONTENT_TYPE_JSON]:
             scoring_response = endpoint.invoke(df, content_type)
             assert scoring_response.status_code == 200, (

